docs/docs/design/network-syncing.md

```diff
@@ -0,0 +1,209 @@
+	AcceptUnreservedPeers()
+	// Disconnect from unreserved peers and deny new unreserved peers to connect for syncing purposes.
+	DenyUnreservedPeers()
+	// Adds a PeerID and its MultiAddr as reserved for a sync protocol (default peer set).
+	//
+	// Returns an error if the given string is not a valid multiaddress or contains an invalid peer ID (which includes
+	// the local peer ID).
+	AddReservedPeer(peer config.MultiaddrPeerId) error
+	// Removes a [peerid.PeerID] from the list of reserved peers for a sync protocol (default peer set).
+	RemoveReservedPeer(peerID peerid.PeerID)
+	// Sets the reserved set of a protocol to the given set of peers.
+	//
+	// Each Multiaddr must end with a "/p2p/" component containing the peer id. It can also
+	// consist of only "/p2p/<peerid>".
+	//
+	// The node will start establishing/accepting connections and substreams to/from peers in this set, if it doesn't
+	// have any substream open with them yet.
+	//
+	// Note however, if a call to this function results in less peers on the reserved set, they will not necessarily
+	// get disconnected (depending on available free slots in the peer set). If you want to also disconnect those
+	// removed peers, you will have to call RemoveFromPeersSet on those in addition to updating the reserved set. You
+	// can omit this step if the peer set is in reserved only mode.
+	//
+	// Returns an error if one of the given addresses is invalid or contains an invalid peer ID (which includes the
+	// local peer id).
+	SetReservedPeers(protocol network.ProtocolName, peers map[multiaddr.Multiaddr]struct{}) error
+	// Add peers to a peer set.
+	//
+	// Each Multiaddr must end with a "/p2p/" component containing the peer id. It can also consist of only
+	// "/p2p/<peerid>".
+	//
+	// Returns an error if one of the given addresses is invalid or contains an invalid peer id (which includes the
+	// local peer id).
+	AddPeersToReservedSet(protocol network.ProtocolName, peers map[multiaddr.Multiaddr]struct{}) error
+	// Remove peers from a peer set.
+	RemovePeersFromReservedSet(protocol network.ProtocolName, peers []peerid.PeerID)
+	// Add a peer to a set of peers.
+	//
+	// If the set has slots available, it will try to open a substream with this peer.
+	//
+	// Each Multiaddr must end with a "/p2p/" component containing the peer id. It can also consist of only
+	// "/p2p/<peerid>".
+	//
+	// Returns an error if one of the given addresses is invalid or contains an invalid peer id (which includes the
+	// local peer id).
+	AddToPeersSet(protocol network.ProtocolName, peers map[multiaddr.Multiaddr]struct{}) error
+	// Remove peers from a peer set.
+	//
+	// If we currently have an open substream with this peer, it will soon be closed.
+	RemoveFromPeersSet(protocol network.ProtocolName, peers []peerid.PeerID)
+	// Returns the number of peers in the sync peer set we're connected to.
+	SyncNumConnected() uint
```

Comment:
if it is not used anywhere maybe we should remove it from inteface?

Reply:
I'd like to keep it in for future work, but the intention is to keep it unimplemented in the translation shim.